src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -2383,6 +2387,23 @@ def _convert_to_snake_case(element):
     return element
 
 
+def get_by_target_resources(cmd, parameters):
+    target_resources_dict = []
+    for target_resource in parameters:
+        target_resources_dict.append({"id": target_resource.id})
+
+    return _DNSListReferences(cli_ctx=cmd.cli_ctx)(command_args={
+        "target_resources": target_resources_dict,
+    })
```

Comment:
It's not a good practice to use custom function to wrap it, recommended to use inherent way. E.g., https://github.com/Azure/azure-cli/blob/82358037c5af4a43c46facfcff054100a730c678/src/azure-cli/azure/cli/command_modules/network/custom.py#L3352. Similar in other places.

Reply:
I used custom wrapper for Create, Update and Delete APIs because they have some deprecated arguments which is not supported by AAZ.